src/OptimalTransport.jl

```diff
@@ -192,7 +199,11 @@ function sinkhorn(mu, nu, C, eps; kwargs...)
     # compute dual potentials
     u, v = sinkhorn_gibbs(mu, nu, K; kwargs...)
 
-    return Diagonal(u) * K * Diagonal(v)
+    if size(mu, 2) == 1
+        return Diagonal(reshape(u, :)) * K * Diagonal(reshape(v, :))
+    else
+        return [Diagonal(u[:, i]) * K * Diagonal(v[:, i]) for i in 1:size(mu, 2)]
+    end
```

Comment:
This creates a type instability: it is not possible to infer from the type of the inputs the type of the output of the function.

Reply:
Thanks for pointing that out. One alternative would be to use instead `mu isa Matrix` to resolve this issue.

Reply:
Hang on, I'm not sure that works :/ Let me take another look at this